pyomo/contrib/iis/tests/test_iis.py

```diff
@@ -0,0 +1,75 @@
+import pyomo.common.unittest as unittest
+import pyomo.environ as pyo
+from pyomo.contrib.iis import write_iis
+from pyomo.common.tempfiles import TempfileManager
+
+import os
+
+
+def _get_infeasible_model():
+    m = pyo.ConcreteModel()
+    m.x = pyo.Var(within=pyo.Binary)
+    m.y = pyo.Var(within=pyo.NonNegativeReals)
+
+    m.c1 = pyo.Constraint(expr=m.y <= 100.0 * m.x)
+    m.c2 = pyo.Constraint(expr=m.y <= -100.0 * m.x)
+    m.c3 = pyo.Constraint(expr=m.x >= 0.5)
+
+    m.o = pyo.Objective(expr=-m.y)
+
+    return m
+
+
+class TestIIS(unittest.TestCase):
+    @unittest.skipUnless(
```

Comment:
The tests look great. I would just recommend adding one more test for the case where the solver is not specified:

```
  @unittest.skipUnless(
         (pyo.SolverFactory("cplex_persistent").available(exception_flag=False) or
          pyo.SolverFactory("gurobi_persistent").available(exception_flag=False) or
          pyo.SolverFactory("xpress_persistent").available(exception_flag=False)),
         "Persistent solver not available",
     )
     def test_write_iis_any_solver(self):
         _test_iis(None)
```

I think this should work without changing any of the other test code???